xncml/parser.py

```diff
@@ -579,7 +675,8 @@ def build_scalar_variable(var_name: str, values_tag: Values, var_type: str) -> x
             ' <values> is empty. Provide a single values within <values></values>'
             ' to preserve the type.'
         )
-        return xr.Variable(data=None, dims=())
+        default_value = nctype(var_type)()
```

Comment:
I changed (again) the behavior of empty scalar parsing here.
For context, here the ncml describe a scalar with a certain type but without providing any value. We would ideally like to create a placeholder for a scalar variable with a numpy type.
But numpy doesn't allow this for scalar, only with an ndarray we can create a typed empty array of a certain shape.

I first though it would be better to loose the type and create a empty scalar with the value `None` but not having this type can mess with subsequent processing.
Now I think it's better to preserve the dtype and fill the scalar with the default value of this dtype.

I would appreciate comments/suggestions here.